mava/systems/tf/maddpg/system.py

```diff
@@ -226,11 +227,14 @@ def __init__(  # noqa
 
         if type(network_sampling_setup) is not list:
             if network_sampling_setup == enums.NetworkSampler.fixed_agent_networks:
-                # if no network_sampling_setup is fixed, use shared_weights to
-                # determine setup
+                # if no network_sampling_setup is specified, assign a single network
+                # to all agents of the same type if weights are shared
+                # else assign seperate networks to each agent
                 self._agent_net_keys = {
-                    agent: "network_0" if shared_weights else f"network_{i}"
-                    for i, agent in enumerate(agents)
+                    agent: f"network_{agent.split('_')[0]}"
+                    if shared_weights
+                    else f"network_{agent}"
+                    for agent in agents
```

Comment:
This seems to actually change network keys from what they were before, no?

Previously, they would be `network_{i}` for each agent index `i`. But now they're `network_{agent}` for each `agent`.

Unsure if this is intentional, and if it is, will it not cause problems in other places where we assume a specific agent net key name convention?

Reply:
Yes, it does change the naming convention but this was not standardised in the past i.e. PPO was different from MADQN & MADDPG. I have standardised it now. There is only a naming convention assumed for agents_ids and not network_ids (from my investigation into the repo)